// Copyright 2018 REMME
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ------------------------------------------------------------------------

syntax = "proto3";

message PermissionMethod {
    enum Method {
        CREATE_DOCUMENT = 0;
        UPDATE_DOCUMENT = 1;
        READ_DOCUMENT = 2;
        CREATE_ACCESS = 3;
        UPDATE_LIST_ACCESS = 4;
    }
}

message Access {
    string pub_container_key = 1;
    string access_key = 2;
}

message Document {
    string id = 1;
    string data = 2;
    repeated Access accesses = 3;
}

message PermissionProtocol {
    string pub_container_key = 1;
    string access_key = 2;
    string data = 3;
    string document_id = 4;
    string grant_pub_container_key = 5;

    repeated Access accesses = 6;
}
